active/app/src/components/worm-grid.js

```diff
@@ -14,6 +14,50 @@ function WormGrid(props) {
     const [limit, setLimit] = useState(30);
     const [loading, setLoading] = useState(false);
 
+
+    // infinite scroll
+    async function fetchMoreData() {
+        const route = `http://localhost:8080/worms?${props.query}&page=${page}&limit=${limit}`;
+        console.log("FETCH MORE DATA params", props.query);
+        let response = await fetch(route)
+        response = await response.json()
+        let data_ = [...data, ...response];
+        await updateData((prevData) => [...prevData, ...response]);
+        await calcNumResultsPerOrder(data_);
+        setLoading(false)
+        if(response.length > 0){
+            window.addEventListener("scroll", handleScroll);
+        }
+    }
+
+    // new filter fetch 
+    async function fetchWithNewFilter() {
+        const route = `http://localhost:8080/worms?${props.query}&page=${1}&limit=${limit}`;
+        let response = await fetch(route);
+        let _data = await response.json();
+        await updateData(_data);
+        await calcNumResultsPerOrder(_data);
+        setLoading(false)
+        if(response.length > 0){
+            window.addEventListener("scroll", handleScroll);
+        }
+    }
```

Comment:
good candidate for refactor into one function

Reply:
I'll add that to the backlog. Good enough for now.